Glossary page (#265)
graphprotocol/docs: pages/en/glossary.mdx

---
title: Glossary
---

- **The Graph**: A decentralized protocol for indexing and querying data.

- **Query**: A request for data. In the case of The Graph, a query is a request for data from a subgraph that will be answered by an Indexer.

- **GraphQL**: A query language for APIs and a runtime for fulfilling those queries with your existing data. The Graph uses GraphQL to query subgraphs.

- **Subgraph**: A custom API built on blockchain data that can be queried using [GraphQL](https://graphql.org/). Developers can build, deploy and publish subgraphs to The Graph's decentralized network. Then, Indexers can begin indexing subgraphs to make them available to be queried by subgraph consumers.

> - **Hosted Service**: A temporary scaffold service for building and querying subgraphs while The Graph's decentralized network matures its cost of service, quality of service, and developer experience.

Reviewer: maybe "as The Graph's decentralized network is maturing.."
- On that note, isn't the Quality of Service better on decentralized network than on hosted? (I think we say so in the docs)

Author: I don't think we say it's better anywhere public yet. Closest we come is [here](https://thegraph.com/docs/en/network-transition-faq/#how-does-quality-of-service-currently-compare-between-the-hosted-service-and-the-decentralized-network), but it's not necessarily saying the QoS is better, but mostly comparable.

> - **Indexers**: Network participants that run indexing nodes to serve queries to subgraphs.

Reviewer: "that run indexing nodes to index data from the blockchains and serve GraphQL queries"

Author: Great addition, thank you!

- **Indexer Revenue Streams**: Indexers are rewarded in GRT with two components: query fee rebates and indexing rewards.

  1. **Query Fee Rebates**: Payments from subgraph consumers for serving queries on the network.

  2. **Indexing Rewards**: The rewards that Indexers receive for indexing subgraphs. Indexing rewards are generated via new issuance of 3% GRT annually.

- **Indexer's Self Stake**: The amount of GRT that Indexers stake to participate in the decentralized network. The minimum is 100,000 GRT, and there is no upper limit.

> - **Delegators**: Network participants who own GRT and stake (delegate) their GRT to Indexers. This allows Indexers to increase their stake in subgraphs on the network. In return, Delegators receive a portion of the Indexing Rewards that Indexers receive for processing subgraphs.

Reviewer: I wouldn't use the word "stake" here since it's confusing

Author: Good point, removed it!

- **Delegation Tax**: A 0.5% fee paid by Delegators when they delegate GRT to Indexers. The GRT used to pay the fee is burned.

- **Curators**: Network participants that identify high-quality subgraphs, and “curate” them (i.e., signal GRT on them) in exchange for curation shares. When Indexers claim query fees on a subgraph, 10% is distributed to the Curators of that subgraph. Indexers earn indexing rewards proportional to the signal on a subgraph. We see a correlation between the amount of GRT signalled and the number of Indexers indexing a subgraph.

- **Curation Tax**: A 1% fee paid by Curators when they signal GRT on subgraphs. The GRT used to pay the fee is burned.

- **Subgraph Consumer**: Any application or user that queries a subgraph.

- **Subgraph Developer**: A developer who builds and deploys a subgraph to The Graph's decentralized network.

> - **Subgraph Manifest**: A JSON file that describes the subgraph's GraphQL schema, data sources, and other metadata.

Reviewer: I would add "here is an example" and link to an example on IPFS

Author: Good call!

- **Rebate Pool**: An economic security measure that holds query fees paid by subgraph consumers until they may be claimed by Indexers as query fee rebates. Residual GRT is burned.

> - **Epoch**: A unit of time that in network. One epoch is currently 6646 blocks. 

Reviewer: or approximately 1 day

Author: Nice catch!

> - **Allocation**: An Indexer can allocate self and delegated stake towards subgraphs that have been published on The Graph's decentralized network. Allocations exist in one of four phases.

Reviewer: "can allocate self and delegated stake" - sounds a bit weird. "An indexer can allocate their total GRT stake (including delegators stake) " - smth like that

Author: THANK YOU! Tried rewording this one a few times and couldn't get it right.

>   1. **Active**: An allocation is considered active when it is created on-chain. This is called opening an allocation, and indicates to the network that the Indexer is actively indexing and serving queries to the subgraph. Active allocations accrue indexing rewards proportional to the signal on the subgraph, and the amount of GRT allocated. 

Reviewer: "to the subgraph" - isn't this "for a particular subgraph"

Author: Amazing catch!

---
title: Glossary
---

> - **The Graph**: A decentralized protocol for indexing and querying data from blockchains.

Reviewer 2: Not only blockchains.

Author: Got it!

> - **Query**: A request for data from a database. In the case of The Graph, a query is a request for data from a subgraph that will be answered by an Indexer.

Reviewer 2: I'd skip the database part. That's a general explanation but a query can be much more than just a request to a database.

Author: Didn't know that, thank you!

- **GraphQL**: A query language for APIs and a runtime for fulfilling those queries with your existing data. The Graph uses GraphQL to query subgraphs.

> - **Subgraph**: A custom API built on blockchain data that can be queried using the [GraphQL](https://graphql.org/). Developers can build, deploy and publish subgraphs to The Graph's decentralized network. Then, Indexers can begin indexing subgraphs to make them available to be queried by subgraph consumers.

Reviewer 3: ```suggestion
- **Subgraph**: A custom API built on blockchain data that can be queried using [GraphQL](https://graphql.org/). Developers can build, deploy and publish subgraphs to The Graph's decentralized network. Then, Indexers can begin indexing subgraphs to make them available to be queried by subgraph consumers.
```

> - **Hosted Service**: A temporary scaffold service for building and querying subgraphs while The Graph's decentralized network matures the cost of service, quality of service, and developer experience.

Reviewer 3: Not sure if better or not, but what do you think of changing "matures the cost of service..." to "matures _its_ cost of service..."?

Author: I like it, thank you!

- **Indexers**: Network participants that run indexing nodes to serve queries to subgraphs.

> - **Indexer revenue streams**: Indexer's are rewarded in GRT with two components: indexing rewards and query fee rebates.

Reviewer 3: ```suggestion
- **Indexer revenue streams**: Indexers are rewarded in GRT with two components: indexing rewards and query fee rebates.
```

  1. **Query fee rebates**: Payments from subgraph consumers for serving queries on the network.

>   2. **Indexing rewards**: The rewards that Indexers receive for indexing subgraphs. Indexing rewards are generated via new issuance of 3% GRT annually.

Reviewer 3: These are listed in the reverse order as mentioned in the previous sentence ("indexing rewards and query fee rebates").

- **Indexer's self stake**: The amount of GRT that Indexers stake to process subgraphs on the network. The minimum is 100,000 GRT, and there is no upper limit.

> - **Delegators**: Individuals who own GRT and stake (delegate) their GRT to Indexers. This allows Indexers to increase their stake in subgraphs on the network. In return, Delegators receive a portion of the Indexing Rewards that Indexers receive for processing subgraphs.

Reviewer 3: The definition of Indexers started with "Network participants that...", and I think it might be good to repeat that for Delegators and Curators.

Author: Good call! Fixed all roles accordingly! Thank you!

- **Delegation tax**: A 0.5% fee paid by Delegators when they delegate GRT to Indexers. The GRT used to pay the fee is burned.

> - **Curators**: Curators identify high-quality subgraphs, and “curate” them (i.e., signaling GRT on them) to earn curation shares, which guarantee a percentage of all future query fees generated by the subgraph. Curating a subgraphs increases the chances of a subgraph being indexed by an Indexer.

Reviewer 3: ```suggestion
- **Curators**: Curators identify high-quality subgraphs, and “curate” them (i.e., signal GRT on them) to earn curation shares, which guarantee a percentage of all future query fees generated by the subgraph. Curating a subgraph increases its chances of being indexed by an Indexer.
```

- **Curation tax**: A 1% fee paid by Curators when they signal GRT on subgraphs. The GRT used to pay the fee is burned.

> - **Subgraph consumers**: Any application or user that queries a subgraph.

Reviewer 3: ```suggestion
- **Subgraph consumer**: Any application or user that queries a subgraph.
```

- **Allocation**: An Indexer can allocate self and delegated stake towards subgraphs that have been published on The Graph's decentralized network. This is called opening an allocation, and indicates to the network that the Indexer is actively indexing and serving queries to the subgraph. Open allocations accrue indexing rewards proportional to the signal on the subgraph, and the amount of GRT allocated. An Indexer may claim the accrued indexing rewards on a given subgraph by submitting a recent, and valid, Proof of Indexing (POI). This is known as closing an allocation. The maximum allocation period is 28 epochs. If an indexer leaves an allocation open beyond 28 epochs, it is known as a stale allocation. Allocations exist in one of four phases:

  1. **Active**: An allocation is considered active when it is created on-chain. The Indexer agent manages creating allocations based on the Indexer rules.

  2. **Closed**: An Indexer can close an allocation once an epoch has passed, or their Indexer agent will automatically close the allocation after ~26 days. When an allocation is closed, their indexing rewards are distributed to the Indexer and its Delegators.

>   3. **Finalized**: Once an allocation has been closed, there is a dispute period, allowing an opportunity for a fisherman to open a dispute.

Reviewer 3: This is not 100% clear to me: can fishermen open a dispute in the "finalized" phase, or _before_ that phase?

Author: Cleared that up! Thank you for pointing that out!

  4. **Claimed**: The final phase of an allocation, all eligible rewards have been distributed and query fee rebates have been claimed.

- **Subgraph Studio**: A powerful dapp for building, deploying, and publishing subgraphs.

- **Fishermen**: Network participants that monitor query results served by Indexers. Fishermen are rewarded with GRT and receive 50% of the Indexer's indexing rewards for that epoch if they find an Indexer serving false data.

- **Arbitrators**: Arbitrators guarantee network security by resolving disputes between Indexers and Fishermen.

> - **Slashing**: Indexers can have their rewards slashed for serving false data. The Indexer's self stake is slashed by 2.5% and the entirety of the rewards for that epoch are slashed as well. Additionally, 50% of the Indexer’s indexing rewards for that epoch are burned, and the other 50% are rewarded to the Fisherman.

Reviewer 3: > the entirety of the rewards for that epoch are slashed as well. Additionally, 50% of the Indexer’s indexing rewards for that epoch are burned, and the other 50% are rewarded to the Fisherman

Is it really "additionally"? Because it seems like we're talking about the same rewards: they are slashed, because 50% of it is burned and 50% goes to the fisherman. No?

Author: Agreed, doesn't make much sense. Removed it!

> - **Indexing Rewards**: The rewards that Indexers receive for processing subgraphs. Indexing rewards are paid out in GRT.

Reviewer 3: Nitpick, but it looks like sometimes we use Title Case (like here), sometimes not (like "Indexer revenue streams", "Delegation tax", etc.)

Author: Thanks for pointing that out! Not a nitpick at all, it was completely random before. Better to capitalize everything, since everything is something specific to the protocol.

- **Delegation Rewards**: The rewards that Delegators receive for delegating GRT to Indexers. Delegation rewards are paid out in GRT.

> - **GRT**: The Graph's work utility token. GRT is used for providing economic incentives to network participants for contributing to the network.

Reviewer 3: ```suggestion
- **GRT**: The Graph's work utility token. GRT provides economic incentives to network participants for contributing to the network.
```
I felt like there were too many fors. :P

Author: You were right!

- **Indexing Node**: A server that runs an Indexer's indexer-agent software. Indexing nodes process subgraphs and make them available to be queried by subgraph consumers.

- **Indexing Node Operators**: The people who run indexing nodes.

- **POI or Proof of Indexing**: POIs are used to verify that an Indexer is indexing the subgraphs they have allocated on.

> - **Indexer-Agent**: The software that runs on an Indexing Node. It processes subgraphs and makes them available to be queried by subgraph consumers.

Reviewer 3: A bit repetitive from "Indexing Node", but maybe that's fine.

Author: Imo, I'd rather be redundant in this doc since this isn't something that people will be reading through, but skimming for what they need - with both terms in here, they'll be able to find either term they're looking for.

- **Graph Explorer**: A dapp designed for network participants to explore subgraphs and interact with the protocol.

> - **Graph CLI**: A command line interface tool for building and deploying to The Graph.

Reviewer 3: Maybe we should add [Graph Client](https://thegraph.com/docs/en/querying/querying-from-an-application/#graph-client) as well?

Author: Good call! Added that

- **Signal Pool**: The on-chain contract that holds the GRT that Curators stake to signal their support for subgraphs.

> - **Cooldown period**: The time remaining until the Indexer can change delegation parameters. Cooldown periods are set up by Indexers when they update their delegation parameters.

Reviewer 3: ```suggestion
- **Cooldown period**: The time remaining until an Indexer who changed their delegation parameters can do so again.
```
(just a suggestion, seems a bit clearer to me)

Author: Great one! Yes, a lot clearer. Thank you :)
